pkg/infrastructure/gcp/clusterapi/dns.go

```diff
@@ -144,10 +150,113 @@ func createDNSRecords(ctx context.Context, client *gcpic.Client, ic *installconf
 	return nil
 }
 
+// createPSCRecords will create and publish the records for the private service connect managed zone.
+func createPSCRecords(ctx context.Context, ic *installconfig.InstallConfig, clusterID string) error {
+	ctx, cancel := context.WithTimeout(ctx, time.Minute*2) // 1 minute for each record below
+	defer cancel()
+
+	computeOpts := []option.ClientOption{option.WithScopes(compute.CloudPlatformScope)}
+	dnsOpts := []option.ClientOption{option.WithScopes(dns.CloudPlatformScope)}
+	pscEndpoint := ic.Config.GCP.Endpoint
+	if pscEndpoint != nil && pscEndpoint.ClusterUseOnly != nil && !(*pscEndpoint.ClusterUseOnly) {
+		computeOpts = append(computeOpts, gcpic.CreateEndpointOption(pscEndpoint.Name, gcpic.ServiceNameGCPCompute))
+		dnsOpts = append(dnsOpts, gcpic.CreateEndpointOption(pscEndpoint.Name, gcpic.ServiceNameGCPDNS))
+	}
+
+	computeService, err := gcpic.GetComputeService(ctx, computeOpts...)
+	if err != nil {
+		return fmt.Errorf("failed to create Compute service: %w", err)
+	}
+
+	endpoint, err := gcpic.GetPrivateServiceConnectEndpoint(computeService, ic.Config.GCP.ProjectID, ic.Config.GCP.Endpoint)
+	if err != nil {
+		return fmt.Errorf("failed to find private service connect endpoint: %w", err)
+	}
+	if endpoint == nil {
+		return fmt.Errorf("private service connect endpoint is nil")
```

Comment:
```suggestion
		return fmt.Errorf("private service connect endpoint %q not found in project %s",
          ic.Config.GCP.Endpoint.Name, ic.Config.GCP.ProjectID)
```

just a bit more information for easy debug :D